app/src/main/java/com/duckduckgo/app/survey/api/SurveyDownloader.kt

```diff
@@ -86,7 +87,9 @@ class SurveyDownloader @Inject constructor(
             }
 
             newSurvey?.let {
-                surveyDao.insert(newSurvey)
+                if (surveyRepository.isUserEligibleForSurvey(newSurvey)) {
+                    surveyDao.insert(newSurvey)
```

Comment:
Seems odd to have a survey repository and then have to also go through the DAO

Reply:
I do it through the Repository to have an extra data layer for Survey. The way I always worked was not letting ViewModels, Presenters, etc have the Dao directly injected but having the feature UseCase or Repository as an intermediate. I know the Dao is still injected in some classes as I only added here the functions needed for the new implementation or small refactor I made. 
The idea would be to refactor survey feature and use the Repository as a public Api instead accessing the Dao directly. Does it make sense @aitorvs ?